noop/noop.go

```diff
@@ -0,0 +1,96 @@
+// Package noop is a Logger that doesn't really log anything
+package noop
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/darvaza-proxy/slog"
+)
+
+var (
+	_ slog.Logger = (*Logger)(nil)
+)
+
+// Logger implements slog.Logger but doesn't log anything
+type Logger struct {
+	level slog.LogLevel
+}
+
+// Enabled tells that we only handle Fatal
+func (nl *Logger) Enabled() bool {
+	if nl == nil || nl.level != slog.Fatal {
+		return false
+	}
+	return true
+}
+
+// WithEnabled passes the logger, but also indicates if it's enabled or not.
+// This logger is only enabled for Fatal entries
+func (nl *Logger) WithEnabled() (slog.Logger, bool) {
+	return nl, nl.Enabled()
+}
+
+// Print pretends to add a log entry with arguments handled in the manner of fmt.Print
+func (nl *Logger) Print(args ...any) {
+	if nl.Enabled() {
```

Comment:
I would put his line in the print function level instead of all the functions that call print

Reply:
It was to avoid formatting strings unnecessarily

Reply:
true, string formatting is quite costly